File: engine/src/main/java/ru/vertuos/engine/world/handlers/updater/VelocityUpdater.java
package ru.vertuos.engine.world.handlers.updater;

import ru.vertuos.engine.world.object.DynamicGameObject;

public class VelocityUpdater implements Updater<DynamicGameObject> {

    @Override
    public void update(DynamicGameObject object, float dt) {
        object.setLinearVelocityX(
            object.getLinearVelocityX() + object.getAccelerationX() * dt
        );
        object.setLinearVelocityY(
            object.getLinearVelocityY() + object.getAccelerationY() * dt
        );
    }
}

File: engine/src/main/kotlin/ru/vertuos/engine/world/updaters/updater/VelocityUpdater.kt
package ru.vertuos.engine.world.updaters.updater

import ru.vertuos.engine.math.plus
import ru.vertuos.engine.math.times
import ru.vertuos.engine.world.obj.DynamicGameObject

class VelocityUpdater : Updater<DynamicGameObject> {

    override fun update(obj: DynamicGameObject, dt: Float) {
        obj.linearVelocity += obj.acceleration * dt
    }
}
